### DISCUSSION TITLE
esp32-2432s028 with slint ?

### DISCUSSION THREAD

**Discussion by ds2k5** [slint-ui/slint#6367] [General]
Hi,
search for a possibility to build a GUI to esp32-2432s028 with Rust Lang.
Is there a example Code ?
Thanks

**Comment by ogoffart** [comment]
I am not familiar with this particular board, but it very likely is.

What you have to do is:
 - check out our docs for Slint on MCU https://docs.rs/slint/latest/slint/docs/mcu/index.html
 - You will need to use the hal crates from [esp-rs](https://github.com/esp-rs/esp-hal) (maybe you can ask further help about esp specific things there)
 - Googling for this board, it seems it uses ILI9341 for its display which is supported by this display driver: https://github.com/almindor/mipidsi
 - You'll also need a touch screen driver. My quick search didn't find what touch screen controller this board is using, but that shouldn't be too hard to figure it out.

**Reply by ogoffart** [reply]
For an example of platform implementation that use a similar hardware, check out https://github.com/slint-ui/slint/blob/master/examples/mcu-board-support/esp32_s3_box.rs

**Reply by ds2k5** [reply]
@ogoffart 

Thanks, will try!
